### ISSUE TITLE
LT09 Should be consistent regarding one or multi target new line behavior

### ISSUE THREAD

**Issue by g1ddy** [sqlfluff/sqlfluff#6549]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### Description

# Enhancement Proposal: Improve Rule LT09 for Select Targets

## Context

Rule LT09 currently states:

> Select targets should be on a new line unless there is only one select target.
<https://docs.sqlfluff.com/en/stable/reference/rules.html#rule-LT09>

This rule can lead to inconsistent behavior when adding or removing select targets.

## Problem

The number of select targets should not impact how the first select target is validated.
Adding a new select target in a new line should not retroactively negate the validity of the first target/line.
It seems like we're conflating two rules that should be mutually exclusive.

## Proposal

1. **Simplify Rule LT09**: Change the rule to "Select targets should be on a new line".
2. **Introduce a New Rule**: Add a new rule or configuration specifically for "allow first target to coexist with select". This would be the exception rather than the rule.

## Benefits

- **Consistency**: Ensures consistent formatting regardless of the number of select targets.
- **Clarity**: Separates concerns by having distinct rules for different scenarios.

## Request for Feedback

We invite maintainers and the community to provide feedback on this proposal. Are there any potential issues or improvements that can be made?

Thank you for considering this enhancement.

## Current Behavior

The following SQL is valid:

```sql
SELECT a
FROM foo;
```

But as soon as you add a new select target, the rule inconsistently changes the behavior of the first select target, and it must now be written as:

```sql
SELECT
    a,
    b
FROM foo;
```

## Proposed Behavior

The rule should consistently enforce that select targets are on new lines, regardless of the number of targets:

```sql
SELECT
    a
FROM foo;
```

```sql
SELECT
    a,
    b
FROM foo;
```

Alternatively, if you enabled to "allow first target to coexist with select" both statements would be more consistent:

```sql
SELECT a
FROM foo;
```

```sql
SELECT a,
    b
FROM foo;
```

### Dialect

This enhancement would apply to any dialects that multiple select targets.

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by bujocek** [comment]
We also would like that. We always want all targets to be on  a new line and we are not able to configure it in the sqlfluff. 
This is how we write selects.
```
SELECT
  a
FROM
  foo;
```
```
SELECT
  a,
  b
FROM
  foo;
```
If we keep LT09 enabled than all single-target selects are marked as warnings. But if we disable the LT09 than programmers can write various weird select formatting which is also unwanted.
